### PR TITLE
raftstore: ignore admin command that removes the leader.

### PR DESCRIPTION
**Pull request by disksing**
cc @siddontang @BusyJay

### CODE REVIEW

File: tests/raftstore/test_conf_change.rs [hunk 1]
```diff
@@ -337,6 +337,7 @@ fn test_after_remove_itself<T: Simulator>(cluster: &mut Cluster<T>) {
 
     // disable auto compact log.
     cluster.cfg.raft_store.raft_log_gc_threshold = 10000;
+    cluster.cfg.raft_store.allow_remove_leader = true;
```
Comment (BusyJay) [inline tests/raftstore/test_conf_change.rs]: Why not update it in `util.rs`? So that it's on by default.

File: src/raftstore/store/peer.rs [hunk 2]
```diff
@@ -1098,6 +1098,14 @@ impl Peer {
         let change_type = change_peer.get_change_type();
         let peer = change_peer.get_peer();
 
+        if change_type == ConfChangeType::RemoveNode && !self.cfg.allow_remove_leader &&
```
Comment (BusyJay) [inline src/raftstore/store/peer.rs]: Please update the function comment.

File: tests/raftstore/test_conf_change.rs (hunk 1 above)
Comment (disksing) [inline tests/raftstore/test_conf_change.rs]: I think the default value should be false. So I only turn it on for special tests that need to remove leader.

Comment (BusyJay) [inline tests/raftstore/test_conf_change.rs]: Then many tests need to be guaranteed that leader should not be the target of `must_remove_peer`.

**Comment by disksing** [comment]
PTAL @BusyJay

**Comment by siddontang** [comment]
PTAL @javaforfun

**Comment by javaforfun** [comment]
LGTM

**Pull request merged**